routes/soldiers.js

```diff
@@ -0,0 +1,70 @@
+import { addNewSoldier, lookForSoldier, lookForAllSoldiers } from '../database.js';
+
+const newSoldierSchema = {
+  schema: {
+    body: {
+      type: 'object',
+      additionalProperties: false,
+      required: ['id', 'name', 'rank', 'limitations'],
+      properties: {
+        id: { type: 'string' },
+        name: { type: 'string' },
+        rank: { type: 'string' },
+        limitations: { type: 'array' },
+      },
+    },
+  },
+
+};
+
+export default async function soldiersRouter(app) {
+  app.post('/', newSoldierSchema, async (req, res) => {
+    const soldier = req.body;
+    try {
+      const insertionRes = await addNewSoldier({
+        _id: soldier.id,
+        name: soldier.name,
+        rank: soldier.rank,
+        limitations: soldier.limitations,
+      });
+      const soldierInserted = await lookForSoldier({ _id: insertionRes.insertedId });
+      res.status(201).send(soldierInserted);
+    } catch (error) {
+      if (error.code === 11000) {
+        res.status(400).send('soldier not created- this id already exists!');
+      } else {
+        res.status(500).send('soldier not created- server error');
+      }
+      app.log.error(error);
+    }
+  });
+
+  app.get('/:id', async (req, res) => {
+    try {
+      const soldierId = req.params.id;
+      const searchResponse = await lookForSoldier(
+        { _id: String(soldierId) },
+      );
+      if (searchResponse) {
+        res.status(200).send(searchResponse);
+      } else if (searchResponse === null) {
+        res.status(404).send('soldier not found');
+      }
+    } catch {
+      res.status(500).send('server error');
+    }
+  });
+
+  app.get('/', async (req, res) => {
+    const soldierQuery = req.query;
+    if (soldierQuery.id) {
+      soldierQuery._id = soldierQuery.id;
+      delete soldierQuery.id;
```

Comment:
`delete` is extremely dangerous because it will change the actual `req.query` (check it out!).
In this is case it will not effect, but can you fix it to not change the `req.query`?